org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/WListItemRenderer.java

```diff
@@ -681,17 +685,24 @@ else if (event.getTarget() instanceof WListbox && Events.ON_SELECT.equals(event.
 			WListbox table = (WListbox) event.getTarget();
 			if (table.isCheckmark()) {
 				int cnt = table.getRowCount();
-				if (cnt == 0 || !(table.getValueAt(0, 0) instanceof IDColumn))
+				if (cnt == 0 || !(table.getValueAt(0, 0) instanceof IDColumn || table.getValueAt(0, 0) instanceof UUIDColumn))
 					return;
 
 				//update IDColumn
 				tableColumn = m_tableColumns.get(0);
 				for (int i = 0; i < cnt; i++) {
-					IDColumn idcolumn = (IDColumn) table.getValueAt(i, 0);
-					Listitem item = table.getItemAtIndex(i);
-
-					value = item.isSelected();
-					Boolean old = idcolumn.isSelected();
+					Boolean old;
+					if (table.getValueAt(i, 0) instanceof IDColumn) {
+						IDColumn idcolumn = (IDColumn) table.getValueAt(i, 0);
+						Listitem item = table.getItemAtIndex(i);
+						value = item.isSelected();
+						old = idcolumn.isSelected();
+					} else {
+						UUIDColumn uucolumn = (UUIDColumn) table.getValueAt(i, 0);
+						Listitem item = table.getItemAtIndex(i);
+						value = item.isSelected();
+						old = uucolumn.isSelected();
+					}
```

Comment:
I think some of the code here and below can be simplify by adding a SelectableIDColumn interface with single boolean isSelected() method and change both IDColumn and UUIDColumn to implement the new interface.

Reply:
Thanks @hengsin - implemented the SelectableIDColumn and simplified the code for those two methods.